db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -647,7 +616,10 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
       return std::make_pair(Status::OK(), !ready);
     };
 
-    bool resuming_from_bg_err = error_handler_.IsDBStopped();
+    bool resuming_from_bg_err =
+        error_handler_.IsDBStopped() ||
+        (cfds[0]->GetFlushReason() == FlushReason::kErrorRecovery ||
+         cfds[0]->GetFlushReason() == FlushReason::kErrorRecoveryRetryFlush);
```

Comment:
Maybe add an assertion somewhere else in this function as follows:
```
for (const auto* cfd : cfds) {
  assert(cfd->GetFlushReason() == cfds[0]->GetFlushReason());
}
```
iirc, the above assertion should hold for atomic flush.